draft-thomson-http-oblivious.md

```diff
@@ -176,88 +186,135 @@ Plaintext Message {
 }
 ~~~
 
-This structure is then encrypted under a key with a specific identity, forming an encapsulated HTTP
-message, with the following structure:
+An Encapsulated Request is comprised of a length-prefixed key identifier and a
+HPKE-protected request message. HPKE protection includes an encapsulated KEM
+shared secret (or `enc`), plus the AEAD-protected request message. An
+Encapsulated Request is shown in {{fig-enc-request}}. {{request}} describes the
+process for constructing and processing an Encapsulated Request.
 
 ~~~
 Key Identifer {
   Key ID Length (i),
   Key ID (..),
 }
 
-Encapsulated Message {
-  Key Identifier (..)
-  Encrypted Message Length (i),
-  Encrypted Message (..),
+Encapsulated Request {
+  Key Identifier (..),
+  Encapsulated KEM Shared Secret (..),
+  AEAD-Protected Request (..),
 }
 ~~~
+{: #fig-enc-request title="Encapsulated Request"}
 
-The encapsulated message Key ID, as well as the encryption mechanics, are different for requests
-and responses, as described below.
+Responses are bound to responses and so consist only of AEAD-protected content.
+{{response}} describes the process for constructing and processing an
+Encapsulated Response.
 
 ## HPKE Encapsulation of Requests {#request}
 
-Clients encapsulate a request Plaintext Message `msg` with an HPKE public key `pkR`, whose Key Identifier
-is `keyID` as follows:
+Clients encapsulate a request `request` with an HPKE public key `pkR`,
+whose Key Identifier is `keyID` as follows:
+
+1. Compute an HPKE context using `pkR`, yielding `context` and encapsulation
+   key `enc`.
+
+2. Construct additional associated data, `aad`, by prepending a single byte
+   with a value of 0x01 to the key identifier. The key identifier length is not
+   included in the AAD.
```

Comment:
Hmm, I think the keyID as sent on the wire, which includes the length, should be the AAD. 

```suggestion
   with a value of 0x01 to the key identifier. The key identifier length is
   included in the AAD.
```